app/controllers/publishers/registrations_controller.rb

```diff
@@ -0,0 +1,135 @@
+        MailerServices::VerifyEmailEmailer.new(publisher: @publisher).perform
+        render :emailed_authentication_token
+      else
+        Rails.logger.error("Create publisher errors: #{@publisher.errors.full_messages}")
+        flash[:warning] = t(".invalid_email")
+        redirect_to sign_up_publishers_path
+      end
+    end
+
+    # This is the method that is called after the user clicks the "Log In" button
+    # If the user is an existing publisher we will send them a log in link, if they are not
+    # then we provide the ability to create an account by clicking the alert on the page.
+    def update
+      @publisher = Publisher.by_email_case_insensitive(params[:email]).first
+      @publisher_email = params[:email]
+
+      enforce_throttle(throttled: throttle_registration?, path: log_in_publishers_path) and return
+
+      if @publisher
+        MailerServices::PublisherLoginLinkEmailer.new(publisher: @publisher).perform
+      else
+        # Failed to find publisher
+        flash[:alert_html_safe] = t("publishers.registrations.emailed_authentication_token.unfound_alert_html",
+          new_publisher_path: sign_up_publishers_path(email: params[:email]),
+          create_publisher_path: registrations_path(email: params[:email]),
+          email: ERB::Util.html_escape(params[:email]))
+
+        @publisher = Publisher.new
+        return redirect_to log_in_publishers_path
+      end
+
+      render :emailed_authentication_token
+    end
+
+    def expired_authentication_token
+      @publisher = Publisher.find(params[:id])
+    end
+
+    # Used by emailed_authentication_token.html.slim to send a new sign up or log in access email
+    # to the publisher passed through the params
+    def resend_authentication_email
+      @publisher = Publisher.find(params[:id])
+
+      enforce_throttle(throttled: throttle_resend_authentication_email?, path: log_in_publishers_path) and return
+
+      if @publisher.email.blank?
+        MailerServices::VerifyEmailEmailer.new(publisher: @publisher).perform
+        @publisher_email = @publisher.pending_email
+      else
+        @publisher_email = @publisher.email
+        MailerServices::PublisherLoginLinkEmailer.new(publisher: @publisher).perform
+      end
+
+      flash.now[:notice] = t(".done")
+      render(:emailed_authentication_token)
+    end
+
+    private
+
+    def email_existing_publisher(publisher)
+      @publisher = publisher
+      @publisher_email = publisher.email
+      MailerServices::PublisherLoginLinkEmailer.new(publisher: @publisher).perform
+      flash.now[:notice] = t("publishers.registrations.create.email_already_active", email: @publisher_email)
+      render :emailed_authentication_token
+    end
+
+    def enforce_throttle(throttled:, path:)
+      @should_throttle = throttled
+      throttle_is_legit = @should_throttle ? verify_recaptcha(model: @publisher) : true
+      return if throttle_is_legit
+
+      Rails.logger.info("User has been throttled")
```

Comment:
Doesn't identify user... which isn't useful. Perhaps we could include some info on who it is or publish in Sentry.

Reply:
True, but unfortunately there's not really a good way of determining what the user is because the user hasn't signed in. IP Addresses shouldn't be logged. I was just hoping to get a good idea on how many users are actually being hit by this enforce throttle. 🤔 

I don't think we should use sentry for things like this, I'd like to reserve sentry for only reporting errors